### PR TITLE
fix(TFD-2388): Fix NPE on reusing LazyAvroCoder in one JVM.

### PR DESCRIPTION
**Pull request by RyanSkraba**
**What is the current behavior?** (You can also link to an open issue here)

LazyAvroCoder setup causes a NullPointerException if more than one job occurs in a JVM.

**What is the new behavior?**

No longer throws the NPE.

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1258/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1258/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1259/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1259/)

File: core/components-adapter-beam/src/main/java/org/talend/components/adapter/beam/coders/LazyAvroCoder.java [hunk 1]
```diff
@@ -80,9 +80,12 @@ public static void resetSchemaSupplier() {
 
     public static void setSchemaSupplier(Supplier<AvroSchemaHolder> factory) {
         // Ensure that the schema supplier has not already been set.
-        if (LazyAvroCoder.threadSchemaSupplierFactory != null)
+        if (threadSchemaSupplierFactory == null) {
+            threadSchemaSupplierFactory = new ThreadLocal<Supplier<AvroSchemaHolder>>();
+        }
+        if (LazyAvroCoder.threadSchemaSupplierFactory.get() != null) {
```
Comment (lbourgeois) [inline core/components-adapter-beam/src/main/java/org/talend/components/adapter/beam/coders/LazyAvroCoder.java]: Could be an `else if` here as if we went into the if block `LazyAvroCoder.threadSchemaSupplierFactory.get()` will return null

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1261/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1261/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1263/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1263/)

**Pull request merged**